app/shop.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from app.customer import Customer
+
+
+class Shop:
+    def __init__(self, name: str, location: list, products: dict) -> None:
+        self._name = name
+        self._location = location
+        self._products = products
+
+    @classmethod
+    def create_shop(cls, shop_info: dict) -> Shop:
+        try:
+            return cls(
+                shop_info["name"],
+                shop_info["location"],
+                shop_info["products"]
+            )
+        except KeyError as e:
+            print(e, "not exist")

```

Comment:
The error handling in the `create_shop` method catches `KeyError`, but the error message 'not exist' is not very informative. Consider providing more context in the error message, such as which key is missing, to make debugging easier.